### ISSUE TITLE
[PAC] Implement the __ptrauth type qualifier.

### ISSUE THREAD

**Issue by asl** [llvm/llvm-project#83796]
Downstream commit: https://github.com/ahmedbougacha/llvm-project/commit/cc7ba7eb1814e9b254c7d94aa0b78cb0e21acfc5

**Comment by antoniofrighetto** [comment]
Related PR: https://github.com/llvm/llvm-project/pull/100830.

**Comment by ahmedbougacha** [comment]
Yep, resolved by https://github.com/llvm/llvm-project/pull/100830
